src/ua/data_types/attribute_id.rs

```diff
@@ -1,8 +1,45 @@
+use std::hash;
+
 use open62541_sys::UA_AttributeId;
 
 crate::data_type!(AttributeId, UInt32);
 
+crate::enum_variants!(
+    AttributeId,
+    UA_AttributeId,
+    [
+        NODEID,
+        NODECLASS,
+        BROWSENAME,
+        DISPLAYNAME,
+        DESCRIPTION,
+        WRITEMASK,
+        USERWRITEMASK,
+        ISABSTRACT,
+        SYMMETRIC,
+        INVERSENAME,
+        CONTAINSNOLOOPS,
+        EVENTNOTIFIER,
+        VALUE,
+        DATATYPE,
+        VALUERANK,
+        ARRAYDIMENSIONS,
+        ACCESSLEVEL,
+        USERACCESSLEVEL,
+        MINIMUMSAMPLINGINTERVAL,
+        HISTORIZING,
+        EXECUTABLE,
+        USEREXECUTABLE,
+        DATATYPEDEFINITION,
+        ROLEPERMISSIONS,
+        USERROLEPERMISSIONS,
+        ACCESSRESTRICTIONS,
+        ACCESSLEVELEX,
+    ]
+);
+
 impl AttributeId {
+    #[deprecated(note = "use `Self::VALUE` instead")]
```

Comment:
No deprecation needed as suggested in #47 to reduce the maintenance burden prior to releasing 1.0.